packages/ckeditor5-dev-dependency-checker/lib/checkdependencies.js

```diff
@@ -369,89 +368,44 @@ async function findMisplacedDependencies( options ) {
 	 */
-	const foldersContainingProductionCode = [
-		/**
-		 * These folders contain the source code of the packages.
-		 */
-		/[/\\]bin[/\\]/,
-		/[/\\]src[/\\]/,
-		/[/\\]lib[/\\]/,
-		/[/\\]theme[/\\]/,
-
-		/**
-		 * This folder contains the compiled code of the packages. Most of this code is the same
-		 * as the source, but during the build process some of the imports are replaced with those
-		 * compatible with the "new installation methods", which may use different dependencies.
-		 *
-		 * For example, the `ckeditor5/src/core.js` import is replaced with `@ckeditor/ckeditor5-core/dist/index.js`.
-		 *                   ^^^^^^^^^                                       ^^^^^^^^^^^^^^^^^^^^^^^^
-		 */
-		/[/\\]dist[/\\]/
-	];
-
-	for ( const absolutePath of absolutePaths ) {
-		if ( !foldersContainingProductionCode.some( folder => absolutePath.match( folder ) ) ) {
-			continue;
-		}
+	/[/\\]bin[/\\]/,
+	/[/\\]src[/\\]/,
+	/[/\\]lib[/\\]/,
+	/[/\\]theme[/\\]/,
 
-		if ( absolutePath.endsWith( '.ts' ) ) {
-			// Verify kind of imports in TypeScript file.
-			const importKinds = await getImportAndExportKinds( packageName, absolutePath );
-
-			// There is any non type kind of import from that package so not a dev dependency.
-			if ( importKinds.some( importKind => importKind != 'type' ) ) {
-				return false;
-			}
-		} else {
-			// Import from some other file from src/ or theme/ - package is not dev dependency.
-			return false;
-		}
-	}
-
-	// There were no value imports, so it is a dev dependency.
-	return true;
-}
+	/**
+	 * This folder contains the compiled code of the packages. Most of this code is the same
+	 * as the source, but during the build process some of the imports are replaced with those
+	 * compatible with the "new installation methods", which may use different dependencies.
+	 *
+	 * For example, the `ckeditor5/src/core.js` import is replaced with `@ckeditor/ckeditor5-core/dist/index.js`.
+	 *                   ^^^^^^^^^                                       ^^^^^^^^^^^^^^^^^^^^^^^^
+	 */
+	/[/\\]dist[/\\]/
+];
 
 /**
- * Parses TS file from `absolutePath` and returns a list of import and export types from `packageName`.
+ * Checks if a given package is a production dependency, i.e., it's used in build files or their typings.
  *
- * @param {string} packageName
- * @param {string} absolutePath File where a given package has been imported.
- * @returns {Promise.<Array.<string>>} Array of import kinds.
+ * @param {string} packageName Name of the package.
+ * @param {Array.<string>} paths Files where a given package has been imported.
+ * @returns {boolean}
  */
-async function getImportAndExportKinds( packageName, absolutePath ) {
-	const astContent = await depCheck.parser.typescript( absolutePath );
-
-	if ( !astContent || !astContent.program || !astContent.program.body ) {
-		return [];
+function isProductionDependency( packageName, paths ) {
+	if ( packageName.startsWith( '@types/' ) ) {
```

Comment:
What about `@types/*` dependencies? Shouldn't this also be a production dependency?